### PR TITLE
Pipelines

### PR DESCRIPTION
**Pull request by andyjessop**
## What this PR solves / how to test

Taking over #6629. Added some ESLint fixes.

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: Affects only pipelines, without E2E tests
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [x] Cloudflare docs PR(s): https://github.com/cloudflare/cloudflare-docs/pull/16639
  - [ ] Not necessary because:

### CODE REVIEW

File: packages/wrangler/src/pipelines/index.ts [hunk 1]
```diff
@@ -17,26 +16,17 @@ import {
...
-	} catch (err) {
-		if (err instanceof APIError) {
-			if (err.code == 10006) {
-				throw new FatalError(`The R2 bucket [${bucket}] doesn't exist`);
-			}
-		}
-		throw err;
+	const bucket = destination.path.bucket;
+	if (!(await getR2Bucket(accountId, bucket))) {
+		logger.warn("Specified bucket doesn't exist:", bucket);
```
Comment (petebacondarwin) [inline packages/wrangler/src/pipelines/index.ts:29]: Why is this no longer a fatal error?
Is it valid to use a non-existing bucket?

Comment (andyjessop) [reply packages/wrangler/src/pipelines/index.ts]: That crept in from a previous commit. I've reverted the change back to Oliver's original now.

File: .changeset/polite-tigers-end.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Cloudflare Pipelines is a product that handles the ingest of event streams
+into R2. This feature integrates various forms of managing pipelines.
+
+Usage:
+wrangler pipelines create <pipeline> Create a new pipeline
+wrangler pipelines list List current pipelines
+wrangler pipelines show <pipeline> Show a pipeline configuration
+wrangler pipelines update <pipeline> Update a pipeline
+wrangler pipelines delete <pipeline> Delete a pipeline
```
Comment (RamIdeas) [inline .changeset/polite-tigers-end.md:15]: We should differentiate between the command and the description for clarity

```suggestion
`wrangler pipelines create <pipeline>`: Create a new pipeline
`wrangler pipelines list`: List current pipelines
`wrangler pipelines show <pipeline>`: Show a pipeline configuration
`wrangler pipelines update <pipeline>`: Update a pipeline
`wrangler pipelines delete <pipeline>`: Delete a pipeline
```

File: packages/wrangler/src/__tests__/pipelines.test.ts [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+	it("create - should fail a missing bucket", async () => {
+		const requests = mockCreeatR2TokenFailure("bad-bucket");
+		await expect(
+			runWrangler("pipelines create bad-pipeline --r2 bad-bucket")
+		).rejects.toThrowError();
+
+		await endEventLoop();
+
+		expect(std.err).toMatchInlineSnapshot(`
+			"[31mX [41;31m[[41;97mERROR[41;31m][0m [1mThe R2 bucket [bad-bucket] doesn't exist[0m
```
Comment (RamIdeas) [inline packages/wrangler/src/__tests__/pipelines.test.ts:360]: nit: consider using normalizingOutput to remove the ANSI codes before snapshotting

File: packages/wrangler/src/index.ts [hunk 4]
```diff
@@ -600,10 +601,19 @@ export function createCLIParser(argv: string[]) {
...
 	// ai
-	wrangler.command("ai", "🤖 Manage AI models\n", (aiYargs) => {
+	wrangler.command("ai", "🤖 Manage AI models", (aiYargs) => {
 		return ai(aiYargs.command(subHelp));
 	});
 
+	// pipelines
+	wrangler.command(
+		"pipelines",
+		`🚰 Manage Worker Pipelines\n`,
```
Comment (RamIdeas) [inline packages/wrangler/src/index.ts:611]: Is Pipelines starting as GA or open-beta? If the latter, this line needs the colourised "[open beta" decorator (see `wrangler vectorize`)

**Review by RamIdeas — APPROVED** [review]
LGTM just the release status needs checking

File: packages/wrangler/src/index.ts [hunk 5]
```diff
@@ -608,7 +608,7 @@ export function createCLIParser(argv: string[]) {
 	// pipelines
 	wrangler.command(
 		"pipelines",
-		`🚰 Manage Worker Pipelines\n`,
+		`🚰 Manage Worker Pipelines ${chalk.hex(betaCmdColor)("[open beta]")}`,
```
Comment (RamIdeas) [inline packages/wrangler/src/index.ts:611]: Still need the `\n` for (hacky) 'grouping' in --help output

Comment (andyjessop) [reply packages/wrangler/src/index.ts:611]: I think the output is fine. Should it look different to this?

```
wrangler pipelines

🚰 Manage Worker Pipelines [open beta]

COMMANDS
  wrangler pipelines create <pipeline>  Create a new pipeline
  wrangler pipelines list               List current pipelines
  wrangler pipelines show <pipeline>    Show a pipeline configuration
  wrangler pipelines update <pipeline>  Update a pipeline
  wrangler pipelines delete <pipeline>  Delete a pipeline

GLOBAL FLAGS
  -j, --experimental-json-config  Experimental: support wrangler.json  [boolean]
  -c, --config                    Path to .toml configuration file  [string]
  -e, --env                       Environment to use for operations and .env files  [string]
  -h, --help                      Show help  [boolean]
  -v, --version                   Show version number  [boolean]
```

Comment (RamIdeas) [reply packages/wrangler/src/index.ts:611]: Not the subhelp (`wrangler pipelines`) but the top-level help (`wrangler --help`)

**Review by RamIdeas — APPROVED** [review]

**Pull request merged**